### PR TITLE
Pass room_version into `event_from_pdu_json`

### PR DESCRIPTION
**Pull request by richvdh**
It's called from all over the shop, so this one's a bit messy.

~~Based on #6823, #6854.~~

### CODE REVIEW

File: synapse/federation/federation_server.py [hunk 1]
```diff
@@ -234,24 +233,17 @@ def __init__(self, hs):
...
-                format_ver = room_version_to_event_format(room_version)
-            except UnsupportedRoomVersionError:
+            except UnsupportedRoomVersionError as e:
                 # this can happen if support for a given room version is withdrawn,
                 # so that we still get events for said room.
-                logger.info(
-                    "Ignoring PDU for room %s with unknown version %s",
-                    room_id,
-                    room_version,
-                )
```
Comment (erikjohnston) [inline synapse/federation/federation_server.py]: If we're logging it seems useful to include *why* we're ignoring an event?

Comment (richvdh) [reply synapse/federation/federation_server.py]: yes, but `str(e)` will tell us that.

Comment (erikjohnston) [reply synapse/federation/federation_server.py]: But it won't log the room_id or the room version? (`UnsupportedRoomVersionError` doesn't include those things)

Comment (richvdh) [reply synapse/federation/federation_server.py]: https://github.com/matrix-org/synapse/blob/rav/redact_changes/6/synapse/storage/data_stores/main/state.py#L81-L82

Comment (erikjohnston) [reply synapse/federation/federation_server.py]: Huh, ok then. (The uses I have seen don't specify a custom message)

**Pull request merged**